### ISSUE TITLE
minimongo TypeError exception when projecting fields on an array containing nulls.

### ISSUE THREAD

**Issue by jean343** [meteor/meteor#10418]
I have data stored in an array of objects containing nulls,
The data looks like:
```
{
    "_id" : "Ge67KtuQfRSq8yaMc",
    "signatory" : [ 
        {
            "name" : {
                "first_name" : "xxxxxx",
                "last_name" : "xxxxxx",
            }
        }, 
        null, 
        null, 
        {
            "name" : {
                "first_name" : "xxxxxx",
                "last_name" : "xxxxxx",
            }
        }
    ]
}
```
When I do my query with the projection `{"signatory.name": 1}`, I get the error:
```
local_collection.js:881 Uncaught TypeError: Cannot convert undefined or null to object
    at hasOwnProperty (<anonymous>)
    at Object.keys.forEach.key (local_collection.js:881)
    at Array.forEach (<anonymous>)
    at transform (local_collection.js:880)
    at doc.map.subdoc (local_collection.js:875)
    at Array.map (<anonymous>)
    at transform (local_collection.js:875)
    at Object.keys.forEach.key (local_collection.js:890)
    at Array.forEach (<anonymous>)
    at transform (local_collection.js:880)
    at Cursor.doc [as _projectionFn] (local_collection.js:904)
    at _getRawObjects.forEach (cursor.js:142)
    at Array.forEach (<anonymous>)
    at Cursor.forEach (cursor.js:140)
    at Cursor.fetch (cursor.js:76)
    at DataGrid.jsx.createContainer (DataGrid.jsx:189)
```

We can see in the debugger that `transform` will be called with `null`
![image](https://user-images.githubusercontent.com/7070881/51155997-9ef34b00-1847-11e9-9b86-e4055cdbd4be.png)

Then `if (!hasOwn.call(doc, key)) {` fails with a type error.

Note that mongodb supports the query, and filters out the null in the response.
![image](https://user-images.githubusercontent.com/7070881/51156075-014c4b80-1848-11e9-9015-e3dd581653be.png)

Note that it does not fail with a single subscription, I have two subscriptions on the same table.

Thanks!!

**Comment by gbhrdt** [comment]
+1 I can confirm this and just ran into the same issue

**Comment by gbhrdt** [comment]
Unstale

**Comment by filipenevola** [comment]
Hi, I believe this was solved here https://github.com/meteor/meteor/pull/10499

Am I right? This is going to be available on Meteor 1.12.
